components/tabs/src/storage.rs

```diff
@@ -198,13 +201,14 @@ impl TabsStorage {
     ) -> Result<()> {
         let connection = self.open_or_create()?;
         let tx = connection.unchecked_transaction()?;
-        // delete the world - we rebuild it from scratch every sync.
-        tx.execute_batch("DELETE FROM tabs")?;
 
+        // For tabs it's fine if we override the existing tabs for a remote
+        // there can only ever be one record for each client
         for crt in new_remote_tabs {
             tx.execute_cached(
-                "INSERT INTO tabs (payload) VALUES (:payload);",
+                "INSERT OR REPLACE INTO tabs (guid, payload) VALUES (:guid, :payload);",
                 rusqlite::named_params! {
+                    ":guid": &crt.client_id,
                     ":payload": serde_json::to_string(&crt).expect("tabs don't fail to serialize"),
```

Comment:
nano-nit - I think avoiding "payload" here makes sense - it's not a "payload" in sync terms and I got a little confused by how the "modified" timestamp was going to work here as the "payload" doesn't have that field) - just renaming it to "json" or similar would be fine.  Possibly in the schema too maybe? (Sorry I didn't mention this before)